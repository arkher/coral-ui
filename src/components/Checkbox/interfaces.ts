export type CheckboxProps = {
  label: string;
  value: boolean;
  onChange: () => void;
  disabled?: boolean;
  required?: boolean;
};
