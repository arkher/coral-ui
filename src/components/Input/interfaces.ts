import { BoxProps } from '@shopify/restyle';
import { TextInputProps, TextInput } from 'react-native';
import { Theme } from '../../themes/default';

type InputProps = {
  variant: CustomHeightComponent;
} & Partial<{
  placeholder: string;
  icon: string;
  bw: CustomBorderWidth;
  p: CustomSpacing;
}> &
  Partial<BoxProps<Theme>> &
  TextInputProps;

type InputRef = HTMLInputElement &
  TextInput & {
    focus(): void;
    blur(): void;
    error(): void;
    success(): void;
    clearStatus(): void;
  };

export { InputProps, InputRef };
