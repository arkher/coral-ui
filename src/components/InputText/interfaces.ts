type CheckboxProps = {
  width: number;
  isFocused: boolean;
  borderColorCustom: string;
  borderColorFocused: string;
};

export { CheckboxProps };
