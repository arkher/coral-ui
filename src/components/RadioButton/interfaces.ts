import { Dispatch } from 'react';

type CustomOptions = {
  label: string;
  value: string | number | boolean;
  disabled?: boolean;
  required?: boolean;
};

type RadioButtonRef = {
  value: string | number | boolean;
};

type RadioButtonProps = {
  label: string;
  options: CustomOptions[];
  initial?: string | number | boolean;
};

type RadioButtonItemProps = {
  label: string;
  value: string | number | boolean;
  disabled?: boolean;
  required?: boolean;
  selectedValue: string | number | boolean;
  setSelectedValue: Dispatch<React.SetStateAction<string | number | boolean>>;
};

export { RadioButtonProps, RadioButtonItemProps, RadioButtonRef };
