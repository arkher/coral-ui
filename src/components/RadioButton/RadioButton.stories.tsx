import { withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { StoriesView } from '../../stories/StorieView';
// import { storiesOf } from '@storybook/react-native';
import RadioButton from './RadioButton';

export default {
  title: 'Componente/RadioButton',
  component: RadioButton,
};

// Stories
export const BasicRadioButton = (): React.ReactNode => (
  <RadioButton label="label" options={[]} />
);

// Add all stories to RN/Expo storybook
storiesOf('RadioButton-ds', module)
  .addDecorator(getStory => <StoriesView>{getStory()}</StoriesView>)
  .addDecorator(withKnobs)
  .add('Default', () => {
    const radioButtonOptions = [
      { value: 0, label: 'zero' },
      { value: 1, label: 'um' },
      { value: 2, label: 'dois' },
    ];

    return (
      <>
        <RadioButton label="RadioButton Label" options={radioButtonOptions} />
      </>
    );
  });
