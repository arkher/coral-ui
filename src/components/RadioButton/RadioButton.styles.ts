import { useTheme } from '@shopify/restyle';
import { StyleSheet } from 'react-native';
import { Theme } from '../../themes';

interface Props {
  checked: boolean;
  disabled: boolean;
  required: boolean;
}

const useStyles = ({ checked, disabled, required }: Props) => {
  const theme = useTheme<Theme>();

  let borderColor;
  if (disabled) borderColor = theme.colors['neutral-light'];
  else if (checked) borderColor = theme.colors['primary-base'];
  else if (required) borderColor = theme.colors['feedback-error-base'];
  else borderColor = theme.colors['neutral-base'];

  let bg;
  if (disabled) bg = theme.colors['neutral-light'];
  // else if (checked) bg = theme.colors['feedback-info-base'];
  else bg = theme.colors.white;

  return StyleSheet.create({
    radioButtonItem: {
      width: 24,
      height: 24,
      borderWidth: checked ? 5 : 2,
      borderRadius: 12,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: bg,
      borderColor,
    },
  });
};

export default useStyles;
