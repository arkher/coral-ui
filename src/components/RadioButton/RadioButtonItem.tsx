import React, { useEffect, useState } from 'react';
import { TouchableHighlight, TouchableWithoutFeedback } from 'react-native';
import Box from '../Box';
import Text from '../Text/Text';
import { RadioButtonItemProps } from './interfaces';
import useStyles from './RadioButton.styles';

const RadioButtonItem: React.FC<RadioButtonItemProps> = ({
  value,
  label,
  required,
  disabled,
  selectedValue,
  setSelectedValue,
}) => {
  const [isMarked, setIsMarked] = useState(value === selectedValue);

  const styles = useStyles({
    checked: isMarked,
    disabled: !!disabled,
    required: !!required,
  });

  const toogle = () => setSelectedValue(value);

  useEffect(() => {
    setIsMarked(value === selectedValue);
  }, [selectedValue, value]);

  return (
    <Box flexDirection="row" alignItems="center" my="quark">
      <TouchableHighlight style={styles.radioButtonItem} onPress={toogle}>
        <Box />
      </TouchableHighlight>
      <TouchableWithoutFeedback onPress={toogle}>
        <Text ml="nano">{label}</Text>
      </TouchableWithoutFeedback>
    </Box>
  );
};

export default RadioButtonItem;
