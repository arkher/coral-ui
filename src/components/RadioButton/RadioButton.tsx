import React, { forwardRef, useImperativeHandle, useState } from 'react';
import Text from '../Text/Text';
import { RadioButtonProps, RadioButtonRef } from './interfaces';
import RadioButtonItem from './RadioButtonItem';

const RadioButton: React.ForwardRefRenderFunction<
  RadioButtonRef,
  RadioButtonProps
> = ({ label, options, initial = '' }, ref) => {
  const [selectedValue, setSelectedValue] = useState(initial);

  useImperativeHandle(
    ref,
    () => ({
      value: selectedValue,
    }),
    [selectedValue],
  );

  return (
    <>
      <Text fs="md" fontWeight="700" color="neutral-darkest" mb="nano">
        {label}
      </Text>
      {options.map(item => (
        <RadioButtonItem
          key={item.label}
          value={item.value}
          label={item.label}
          required={item.required}
          disabled={item.disabled}
          selectedValue={selectedValue}
          setSelectedValue={setSelectedValue}
        />
      ))}
    </>
  );
};

export default forwardRef(RadioButton);
