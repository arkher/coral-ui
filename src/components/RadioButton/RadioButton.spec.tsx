import React, { createRef } from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react-native';
import { ThemeProvider } from '@shopify/restyle';
import RadioButton from './RadioButton';
import { themeMaestro } from '../../themes';
import { RadioButtonRef } from './interfaces';

afterEach(cleanup);

jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon');

it('should be like initial value', () => {
  const ref = createRef<RadioButtonRef>();

  const radioButtonOptions = [
    { value: 0, label: 'zero' },
    { value: 1, label: 'um' },
  ];

  render(
    <ThemeProvider theme={themeMaestro}>
      <RadioButton
        ref={ref}
        initial={0}
        label="RadioButton Label"
        options={radioButtonOptions}
      />
    </ThemeProvider>,
  );

  expect(ref.current?.value).toBe(0);
});

it('should be changed value', () => {
  const ref = createRef<RadioButtonRef>();

  const radioButtonOptions = [
    { value: 0, label: 'zero' },
    { value: 1, label: 'um' },
    { value: 2, label: 'dois' },
  ];

  const { getByText } = render(
    <ThemeProvider theme={themeMaestro}>
      <RadioButton
        ref={ref}
        initial={0}
        label="RadioButton Label"
        options={radioButtonOptions}
      />
    </ThemeProvider>,
  );

  fireEvent.press(getByText('um'));
  expect(ref.current?.value).toBe(1);

  fireEvent.press(getByText('dois'));
  expect(ref.current?.value).toBe(2);
});

test('should RadioButton render correctly', async () => {
  const radioButtonOptions = [
    { value: 0, label: 'zero' },
    { value: 1, label: 'um' },
  ];

  const { toJSON } = render(
    <ThemeProvider theme={themeMaestro}>
      <RadioButton label="RadioButton Label" options={radioButtonOptions} />
    </ThemeProvider>,
  );

  expect(toJSON()).toMatchSnapshot();
});
