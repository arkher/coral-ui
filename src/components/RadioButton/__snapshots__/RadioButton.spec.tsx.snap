// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should RadioButton render correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#2D3338",
          "display": "flex",
          "fontFamily": "\\"Inter\\", sans-serif",
          "fontWeight": "700",
          "marginBottom": 8,
        },
        Array [
          Object {
            "fontSize": 14,
          },
        ],
      ]
    }
  >
    RadioButton Label
  </Text>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginVertical": 4,
        },
        Array [
          Object {},
        ],
      ]
    }
    testID="Box"
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#B4BBC2",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
            },
            Array [
              Object {},
            ],
          ]
        }
        testID="Box"
      />
    </View>
    <Text
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "display": "flex",
            "fontFamily": "\\"Inter\\", sans-serif",
            "fontWeight": "400",
            "marginLeft": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      zero
    </Text>
  </View>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "marginVertical": 4,
        },
        Array [
          Object {},
        ],
      ]
    }
    testID="Box"
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#B4BBC2",
          "borderRadius": 12,
          "borderWidth": 2,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "display": "flex",
            },
            Array [
              Object {},
            ],
          ]
        }
        testID="Box"
      />
    </View>
    <Text
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "display": "flex",
            "fontFamily": "\\"Inter\\", sans-serif",
            "fontWeight": "400",
            "marginLeft": 8,
          },
          Array [
            Object {},
          ],
        ]
      }
    >
      um
    </Text>
  </View>,
]
`;
